src/api/objects/ObjectAPI.js

```diff
@@ -312,6 +315,27 @@ ObjectAPI.prototype.save = function (domainObject) {
     return result;
 };
 
+/**
+ * After entering into edit mode, creates a new instance of TransactionManager to keep track of changes in Objects
+ */
+ObjectAPI.prototype.startTransaction = function () {
+    this.transactionManager = new TransactionManager();
+};
+
+/**
+ * When in edit mode, after save action commit/persists all stored transactions
+ */
+ObjectAPI.prototype.CommitAllTransactions = function () {
+    return this.transactionManager.CommitAllTransactions(this.save.bind(this));
```

Comment:
Might be cleaner if we just pass the objectAPI into the `TransactionManager`. Having a two way dependency is not as nice as having a one-way dependency, but I still think it's cleaner than passing in a bound function. That suggests a degree of flexibility that just doesn't exist, we're always going to use `ObjectAPI.save()` to persist the objects.